python/lsst/daf/butler/transfers/_yaml.py

```diff
@@ -337,28 +337,23 @@ def load(self, datastore: Optional[Datastore], *,
         # FileDatasets to ingest into the datastore (in bulk):
         fileDatasets = []
         for (datasetTypeName, run), records in self.datasets.items():
-            datasetType = self.registry.getDatasetType(datasetTypeName)
             # Make a big flattened list of all data IDs and dataset_ids, while
             # remembering slices that associate them with the FileDataset
             # instances they came from.
-            dataIds: List[DataCoordinate] = []
+            datasets: List[DatasetRef] = []
             dataset_ids: List[int] = []
             slices = []
             for fileDataset in records:
-                start = len(dataIds)
-                dataIds.extend(ref.dataId for ref in fileDataset.refs)
+                start = len(datasets)
+                datasets.extend(fileDataset.refs)
                 dataset_ids.extend(ref.id for ref in fileDataset.refs)  # type: ignore
-                stop = len(dataIds)
+                stop = len(datasets)
                 slices.append(slice(start, stop))
             # Insert all of those DatasetRefs at once.
             # For now, we ignore the dataset_id we pulled from the file
             # and just insert without one to get a new autoincrement value.
             # Eventually (once we have origin in IDs) we'll preserve them.
-            resolvedRefs = self.registry.insertDatasets(
-                datasetType,
-                dataIds=dataIds,
-                run=run,
-            )
+            resolvedRefs = self.registry.importDatasets(datasets)
```

Comment:
The new method has simplified this so that's good.

Reply:
Comment above this line about "once we have origin IDs" is now out of date.